File: src/f90/prog/ssht_about.f90
! SSHT package to perform spin spherical harmonic transforms
! Copyright (C) 2011  Jason McEwen
! See LICENSE.txt for license details


!------------------------------------------------------------------------------
! ssht_about
!
!> Prints information about the SSHT package, including version
!! and build numbers. 
!!
!! Usage: ssht_about
!!     
!! @author <a href="http://www.jasonmcewen.org">Jason McEwen</a>
!
! Revisions:
!   October 2011 - Written by Jason McEwen 
!------------------------------------------------------------------------------

program ssht_about

  implicit none

  write(*,'(a)') "=========================================================="
  write(*,'(a)') "SSHT package to perform spin spherical harmonic transforms"
  write(*,'(a)') "By Jason McEwen and Yves Wiaux"

  write(*,'(a)') "See www.jasonmcewen.org for more information."
  write(*,'(a)') "See LICENSE.txt for license details."

  write(*,'(a,a)') "Version: ", SSHT_VERSION
  write(*,'(a,a)') "Build: ", SSHT_BUILD
  write(*,'(a)') "=========================================================="

end program ssht_about

File: src/c/ssht_about.c
// SSHT package to perform spin spherical harmonic transforms
// Copyright (C) 2011  Jason McEwen
// See LICENSE.txt for license details

/*! 
 * \file ssht_about.c
 * Print information about the SSHT package, including version
 * and build numbers. 
 *
 * Usage: ssht_about
 *
 * \author <a href="http://www.jasonmcewen.org">Jason McEwen</a>
 */

#include <stdio.h>
#ifdef BUILT_WITH_CMAKE
#include "ssht/ssht_version.h"
#endif

int main(void) {

#ifndef BUILT_WITH_CMAKE
  printf("%s\n", "==========================================================");
  printf("%s\n", "SSHT package to perform spin spherical harmonic transforms");
  printf("%s\n", "By Jason McEwen, Chris Wallis, Martin Buttner, ");
  printf("%s\n", "Boris Leistedt, and Yves Wiaux");

  printf("%s\n", "See www.jasonmcewen.org for more information.");
  printf("%s\n", "See LICENSE.txt for license details.");

  printf("%s%s\n", "Version: ", SSHT_VERSION);
  printf("%s%s\n", "Build: ", SSHT_BUILD);
  printf("%s\n", "==========================================================");
#else
  printf("%s", ssht_info());
#endif

  return 0;

}
